stock_release_channel_shipment_advice/models/stock_release_channel.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import Command, _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class StockReleaseChannel(models.Model):
+    _inherit = "stock.release.channel"
+
+    shipment_planning_method = fields.Selection(
+        selection=[("none", "None"), ("simple", "Simple")], default="none"
+    )
+    picking_to_plan_ids = fields.Many2many(
+        comodel_name="stock.picking", compute="_compute_picking_to_plan_ids"
+    )
+    can_plan_shipment = fields.Boolean(compute="_compute_can_plan_shipment")
+    shipment_advice_ids = fields.One2many(
+        comodel_name="shipment.advice",
+        inverse_name="release_channel_id",
+        string="Shipment Advice",
+        readonly=True,
+    )
+
+    def button_show_shipment_advice(self):
+        self.ensure_one()
+        return {
+            "type": "ir.actions.act_window",
+            "name": _("Shipment Advice"),
+            "view_mode": "calendar,tree,form",
+            "res_model": self.shipment_advice_ids._name,
+            "domain": [("id", "in", self.shipment_advice_ids.ids)],
+            "context": self.env.context,
+        }
+
+    @api.depends("shipment_planning_method", "picking_to_plan_ids")
+    def _compute_can_plan_shipment(self):
+        for rec in self:
+            rec.can_plan_shipment = (
+                rec.picking_to_plan_ids
+                and rec.shipment_planning_method
+                and rec.shipment_planning_method != "none"
+            )
+
+    @api.depends("picking_ids", "picking_ids.planned_shipment_advice_id")
+    def _compute_picking_to_plan_ids(self):
+        for rec in self:
+            rec.picking_to_plan_ids = rec.picking_ids.filtered(
+                lambda p: not p.planned_shipment_advice_id
+            )
+
+    def button_plan_shipment(self):
+        self.ensure_one()
+        if not self.can_plan_shipment:
+            raise UserError(_("Shipment planning not allowed"))
+        shipment_advice = self._plan_shipment()
+        if not shipment_advice:
+            return {}
+        return {
+            "type": "ir.actions.act_window",
+            "name": _("Shipment Advice"),
+            "view_mode": "form",
+            "res_model": shipment_advice._name,
+            "res_id": shipment_advice.id,
+            "context": self.env.context,
+        }
+
+    def _plan_shipment(self):
+        self.ensure_one()
```

Comment:
I don't see why this method should not be implemented to work on a multi records recordset...